### PR TITLE
[IMP] stock: add attribute `_get_default_warehouse_id` to res.users

### PR DESCRIPTION
**Pull request by jepbarasgarov**
After version >= `16.4` module **repair** is **[using](https://github.com/odoo/odoo/blob/c002aad906bef390d6d7fd5c795a8e6baa5fd31d/addons/repair/models/repair.py#L571-L578)** attribute `_get_default_warehouse_id` which is **[defined](https://github.com/odoo/odoo/blob/c002aad906bef390d6d7fd5c795a8e6baa5fd31d/addons/sale_stock/models/res_users.py#L12-L17)** in module `sale_stock`. There is no dependency between that 2 modules.

But as the module `sale_stock` is **autoinstall=1** and its all dependencies are being installed when we install `repair` the module **sale_stock** is also being installed automatically. That is why when we install **repair** we will not have any problem on using that attribute. If we uninstall that `sale_stock` we will have problem because `repair` module will not be able to find the attribute `_get_default_warehouse_id`.

For fixing the issue I added that attribute in module `stock` as it is dependent for both of `repair` and `sale_stock`, and also I use the inheritence in module `sale_stock`

**Steps to reproduce:**

1. Install `repair` on `16.0`
2. Uninstall `sale_stock`
3. Try to upgrade to `17.0`

You will error similar to this:

```
 File "/home/odoo/src/odoo/17.0/addons/repair/models/repair.py", line 28, in _default_picking_type_id
    return self._get_picking_type().get((self.env.company, self.env.user))
  File "/home/odoo/src/odoo/17.0/addons/repair/models/repair.py", line 563, in _get_picking_type
    default_warehouse = self.env.user.with_company(companies.id)._get_default_warehouse_id()
AttributeError: 'res.users' object has no attribute '_get_default_warehouse_id'
```

upg-1565705

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aj-fuentes** [comment]
Add a reference to the commit that introduced the issue in `repair` on your commit message eaa6e1a08c3d2cdcb558689523e5bd332978f5e4

I updated your links in the PR description to be permalinks.

The rest LGTM.

**Comment by jepbarasgarov** [comment]
> Add a reference to the commit that introduced the issue in `repair` on your commit message [eaa6e1a](https://github.com/odoo/odoo/commit/eaa6e1a08c3d2cdcb558689523e5bd332978f5e4)
> 
> I updated your links in the PR description to be permalinks.
> 
> The rest LGTM.

I added the reference to the commit which introduced the issue in `repair`

File: addons/stock/models/__init__.py [hunk 1]
```diff
@@ -20,3 +20,4 @@
 from . import stock_package_level
 from . import stock_package_type
 from . import stock_storage_category
+from . import res_users
```
Comment (ticodoo) [inline addons/stock/models/__init__.py:23]: Can you add this to underneath the `res_partner` import so we can at least try to pretend like we're following alphabetical ordering (I see that it's already a bit of a mess :disappointed: )

File: addons/sale_stock/models/res_users.py [hunk 2]
```diff
@@ -5,16 +5,14 @@
 
 
 class Users(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
```
Comment (ticodoo) [inline addons/sale_stock/models/res_users.py:8]: Probably shouldn't do a superficial change like this in stable due to the unnecessary diff

**Review by ticodoo — COMMENTED** [review]
Fix looks good to me, just a couple of nitpick cleanup things to apply and then I'll r+ it. Thanks!

File: addons/stock/models/__init__.py (hunk 1 above)
Comment (jepbarasgarov) [reply addons/stock/models/__init__.py]: Thanks for pointing it ;)
I have implemented this change

File: addons/sale_stock/models/res_users.py (hunk 2 above)
Comment (jepbarasgarov) [reply addons/sale_stock/models/res_users.py]: Changes have been implemented, thanks ;)

**Comment by ticodoo** [comment]
Thanks!

@robodoo r+
